<!--
   Author: Russell Toris, Worcester Polytechnic Institute
   Author: David Kent, Worcester Polytechnic Institute
  Version: March 31, 2015
-->

<launch>
  <!-- Grasp Database Connection Params -->
  <arg name="host" default="127.0.0.1" />
  <arg name="port" default="5432" />
  <arg name="user" default="ros" />
  <arg name="password" default="" />
  <arg name="db" default="graspdb" />

  <!-- Recognition Listener Params -->
  <arg name="segmented_objects_topic" default="/segmentation/segmented_objects" />
  <arg name="debug" default="false" />

  <!-- Set Global Params -->
  <param name="/graspdb/host" type="str" value="$(arg host)" />
  <param name="/graspdb/port" type="int" value="$(arg port)" />
  <param name="/graspdb/user" type="str" value="$(arg user)" />
  <param name="/graspdb/password" type="str" value="$(arg password)" />
  <param name="/graspdb/db" type="str" value="$(arg db)" />

  <node pkg="rail_recognition" name="object_recognition_listener" type="object_recognition_listener" output="screen">
    <param name="segmented_objects_topic" value="$(arg segmented_objects_topic)" />
    <param name="debug" value="$(arg debug)" />
  </node>
</launch>
